Source/DafnyCore/DafnyMain.cs

```diff
@@ -140,83 +98,22 @@ public class DafnyMain {
-      }
-
-      if (builtIns.Options.PrintIncludesMode != DafnyOptions.IncludesModes.None) {
-        var dependencyMap = new DependencyMap();
-        dependencyMap.AddIncludes(allIncludes);
-        dependencyMap.PrintMap(builtIns.Options);
-      }
-
-      return null; // Success
-
-      string TraverseIncludesFrom(int startingIndex) {
-        var includeIndex = startingIndex;
-        var stack = new Stack<Include>();
-
-        while (true) {
-          var addedItems = allIncludes.Skip(includeIndex);
-          foreach (var addedItem in addedItems.Reverse()) {
-            stack.Push(addedItem);
-          }
-
-          includeIndex = allIncludes.Count;
-
-          if (stack.Count == 0) {
-            break;
-          }
-
-          var include = stack.Pop();
-          if (!includesFound.Add(include) || excludeFiles.Contains(include.CanonicalPath)) {
-            continue;
-          }
-
-          DafnyFile file;
-          try {
-            file = new DafnyFile(builtIns.Options, include.IncludedFilename);
-          } catch (IllegalDafnyFile) {
-            return ($"Include of file \"{include.IncludedFilename}\" failed.");
-          }
-
-          string result = ParseFile(file, include, module, builtIns, errs);
-          if (result != null) {
-            return result;
-          }
-        }
-
-        return null;
-      }
-    }
-
-    private static string ParseFile(DafnyFile dafnyFile, Include include, ModuleDecl module, BuiltIns builtIns, Errors errs) {
-      var fn = builtIns.Options.UseBaseNameForFileName ? Path.GetFileName(dafnyFile.FilePath) : dafnyFile.FilePath;
-      try {
-        int errorCount = Parser.Parse(dafnyFile.Content, dafnyFile.Uri, module, builtIns, errs);
-        if (errorCount != 0) {
-          return $"{errorCount} parse errors detected in {fn}";
-        }
-      } catch (IOException e) {
-        IToken tok = include == null ? Token.NoToken : include.tok;
-        errs.SemErr(tok, "Unable to open included file");
-        return $"Error opening file \"{fn}\": {e.Message}";
-      }
-
-      return null; // Success
-    }
+    // private static string ParseFile(DafnyFile dafnyFile, Include include, BuiltIns builtIns, ErrorReporter errorReporter) {
+    //   var fn = builtIns.Options.UseBaseNameForFileName ? Path.GetFileName(dafnyFile.FilePath) : dafnyFile.FilePath;
+    //   try {
+    //     var parseResult = ParseUtils.Parse(dafnyFile.Content, dafnyFile.Uri, builtIns, errorReporter);
+    //     var errorCount = parseResult.ErrorCount;
+    //     if (errorCount != 0) {
+    //       return $"{errorCount} parse errors detected in {fn}";
+    //     }
+    //   } catch (IOException e) {
+    //     IToken tok = include == null ? Token.NoToken : include.tok;
+    //     errorReporter.Error(MessageSource.Parser, tok, "Unable to open included file");
+    //     return $"Error opening file \"{fn}\": {e.Message}";
+    //   }
+    //
+    //   return null; // Success
+    // }
```

Comment:
Is this still needed?

Reply:
Ugh, woops. Will remove in the follow-up PR (https://github.com/dafny-lang/dafny/pull/4083)